Generic function that accepts any tx, clonning StaticTxPayload tx

I want to write a function where tx is passed as argument and tx is cloned.

   ` let tx: StaticTxPayload<AddCitizen> = polkadot::tx().profile_validation().add_citizen(profile_data);`

Seems clone is not implemented for StaticTxPayload, not sure how to make a generic function that accepts any tx.

Not sure if its subxt question, but while writing a hook for yew it gives error due to TxPayload

```rust
use crate::components::accounts::account_store::PhraseStore;
use crate::components::accounts::functions::get_from_seed_sr;
use std::pin::Pin;
use std::rc::Rc;
use subxt::blocks::ExtrinsicEvents;
use subxt::config::WithExtrinsicParams;
use subxt::tx::BaseExtrinsicParams;
use subxt::tx::PlainTip;
use subxt::SubstrateConfig;
use subxt::{tx::PairSigner, PolkadotConfig};
use yew::prelude::*;
use yewdux::prelude::*;

// pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T>  + 'a>>;

pub struct UseSignTxHandle {
    api_call: Rc<
        dyn Fn() -> Pin<
            Box<
                (dyn std::future::Future<
                    Output = std::option::Option<
                        ExtrinsicEvents<
                            WithExtrinsicParams<
                                SubstrateConfig,
                                BaseExtrinsicParams<SubstrateConfig, PlainTip>,
                            >,
                        >,
                    >,
                > + 'static),
            >,
        >,
    >,
}

impl UseSignTxHandle {
    pub fn api_call(&self) {
        (self.api_call)();
    }
}

impl Clone for UseSignTxHandle {
    fn clone(&self) -> Self {
        Self {
            api_call: self.api_call.clone(),
        }
    }
}

#[hook]
pub fn use_sign_tx_handle<T>(tx: T) -> UseSignTxHandle
where
    T: subxt::tx::TxPayload,
{
    let (store, _) = use_store::<PhraseStore>();
    let store_clone = store.clone();
    let api_call = {
        let phrase_option = store_clone.mnemonic_phrase.clone();
        Rc::new(move || {
            let phrase_option = phrase_option.clone();
            let pin = Box::pin(async move {
                let phrase_option = phrase_option.clone();
                let client =
                    subxt::client::OnlineClient::<PolkadotConfig>::from_url("ws://127.0.0.1:9944")
                        .await
                        .unwrap();
                if let Some(seed) = phrase_option {
                    let pair = get_from_seed_sr(&seed);
                    let signer = PairSigner::new(pair);
                    let result = client
                        .tx()
                        .sign_and_submit_then_watch_default(&tx, &signer)
                        .await
                        .unwrap()
                        .wait_for_finalized()
                        .await
                        .unwrap();
                    let events = result.fetch_events().await.unwrap();
                    Some(events)
                } else {
                    None
                }
            });
            pin as _
        })
    };
    UseSignTxHandle { api_call }
}
```

```rust
let tx = polkadot::tx().profile_validation().add_citizen(profile_data);
let handler = use_sign_tx_handle(tx);
```

Error:
```
error[E0525]: expected a closure that implements the `Fn` trait, but this closure only implements `FnOnce`
  --> src/components/accounts/hooks/sign_tx_handle.rs:58:17
   |
58 |         Rc::new(move || {
   |                 ^^^^^^^ this closure implements `FnOnce`, not `Fn`
...
71 |                         .sign_and_submit_then_watch_default(&tx, &signer)
   |                                                              -- closure is `FnOnce` because it moves the variable `tx` out of its environment
...
86 |     UseSignTxHandle { api_call }
   |                       -------- the requirement to implement `Fn` derives from here
```

I am trying to make a hook in yew:
```
use crate::components::accounts::account_store::PhraseStore;
use crate::components::accounts::functions::get_from_seed_sr;
use std::pin::Pin;
use std::rc::Rc;
use subxt::blocks::ExtrinsicEvents;
use subxt::config::WithExtrinsicParams;
use subxt::tx::TxPayload;
use subxt::tx::BaseExtrinsicParams;
use subxt::tx::PlainTip;
use subxt::SubstrateConfig;
use subxt::{tx::PairSigner, PolkadotConfig};
use yew::prelude::*;
use yewdux::prelude::*;

// pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T>  + 'a>>;

pub struct UseSignTxHandle {
    pub api_call: Rc<
        dyn Fn(&dyn TxPayload) -> Pin<
            Box<
                (dyn std::future::Future<
                    Output = std::option::Option<
                        ExtrinsicEvents<
                            WithExtrinsicParams<
                                SubstrateConfig,
                                BaseExtrinsicParams<SubstrateConfig, PlainTip>,
                            >,
                        >,
                    >,
                > + 'static),
            >,
        >,
    >,
}



impl Clone for UseSignTxHandle {
    fn clone(&self) -> Self {
        Self {
            api_call: self.api_call.clone(),
        }
    }
}

#[hook]
pub fn use_sign_tx_handle() -> UseSignTxHandle
{
    let (store, _) = use_store::<PhraseStore>();
    let store_clone = store.clone();
    let api_call = {
        let phrase_option = store_clone.mnemonic_phrase.clone();
        Rc::new(move |tx: &dyn TxPayload | {
            let phrase_option = phrase_option.clone();
            let pin = Box::pin(async move {
                let phrase_option = phrase_option.clone();
                let client =
                    subxt::client::OnlineClient::<PolkadotConfig>::from_url("ws://127.0.0.1:9944")
                        .await
                        .unwrap();
                if let Some(seed) = phrase_option {
                    let pair = get_from_seed_sr(&seed);
                    let signer = PairSigner::new(pair);
                    let result = client
                        .tx()
                        .sign_and_submit_then_watch_default(tx, &signer)
                        .await
                        .unwrap()
                        .wait_for_finalized()
                        .await
                        .unwrap();
                    let events = result.fetch_events().await.unwrap();
                    Some(events)
                } else {
                    None
                }
            });
            pin as _
        })
    };
    UseSignTxHandle { api_call }
}
```
```rust
wasm_bindgen_futures::spawn_local(async move {
                let tx = polkadot::tx().profile_validation().add_citizen(profile_data);
                let data = (handler.api_call)(&tx).await;
            });
```

It gives error:
```
error[E0277]: the size for values of type `dyn TxPayload` cannot be known at compilation time
   --> src/components/accounts/hooks/sign_tx_handle.rs:66:61
    |
66  |                         .sign_and_submit_then_watch_default(tx, &signer)
    |                          ---------------------------------- ^^ doesn't have a size known at compile-time
    |                          |
    |                          required by a bound introduced by this call
    |
    = help: the trait `Sized` is not implemented for `dyn TxPayload`
```

But works fine with String
```rust

use std::pin::Pin;
use std::rc::Rc;
use yew::prelude::*;


// pub type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T>  + 'a>>;

pub struct UseSignTxHandle {
    pub api_call: Rc<dyn Fn(String) -> Pin<Box<dyn std::future::Future<Output = Option<String>>>>>,
}


impl Clone for UseSignTxHandle {
    fn clone(&self) -> Self {
        Self {
            api_call: self.api_call.clone(),
        }
    }
}


#[hook]
pub fn use_sign_tx_handle() -> UseSignTxHandle

{
    let state: UseStateHandle<Option<String>> = use_state(|| None);
    let api_call = {
        Rc::new(move |txt:String| {
            let state = state.clone();
            Box::pin(async move { Some(txt) }) as _
        })
    };
    UseSignTxHandle { api_call }
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,8 @@ This is a fairly small release, primarily to bump substrate dependencies to thei
 
 The main breaking change is fairly small: [#804](https://github.com/paritytech/subxt/pull/804). Here, the `BlockNumber` associated type has been removed from `Config` entirely, since it wasn't actually needed anywhere in Subxt. Additionally, the constraints on each of those associated types in `Config` were made more precise, primarily to tidy things up (but this should result in types more easily being able to meet the requirements here). If you use custom `Config`, the fix is simply to remove the `BlockNumber` type. If you also use the `Config` trait in your own functions and depend on those constraints, you may be able to define a custom `MyConfig` type which builds off `Config` and adds back any additional bounds that you want.
 
+Note worthy PRs merged since the last release:
+
 ## Added
 
 - Add find last function ([#821](https://github.com/paritytech/subxt/pull/821))
@@ -18,7 +20,6 @@ The main breaking change is fairly small: [#804](https://github.com/paritytech/s
 ## Changed
 
 - Remove unneeded Config bounds and BlockNumber associated type ([#804](https://github.com/paritytech/subxt/pull/804))
-- Remove sleeps from release checklist; no longer needed ([#803](https://github.com/paritytech/subxt/pull/803))
 
 ## [0.26.0] - 2022-01-24
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.27.0] - 2022-02-13
+
+This is a fairly small release, primarily to bump substrate dependencies to their latest versions.
+
+The main breaking change is fairly small: [#804](https://github.com/paritytech/subxt/pull/804). Here, the `BlockNumber` associated type has been removed from `Config` entirely, since it wasn't actually needed anywhere in Subxt. Additionally, the constraints on each of those associated types in `Config` were made more precise, primarily to tidy things up (but this should result in types more easily being able to meet the requirements here). If you use custom `Config`, the fix is simply to remove the `BlockNumber` type. If you also use the `Config` trait in your own functions and depend on those constraints, you may be able to define a custom `MyConfig` type which builds off `Config` and adds back any additional bounds that you want.
+
+## Added
+
+- Add find last function ([#821](https://github.com/paritytech/subxt/pull/821))
+- Doc: first item is current version comment ([#817](https://github.com/paritytech/subxt/pull/817))
+- Remove sleeps from release checklist; no longer needed ([#803](https://github.com/paritytech/subxt/pull/803))
+
+## Changed
+
+- Remove unneeded Config bounds and BlockNumber associated type ([#804](https://github.com/paritytech/subxt/pull/804))
+
 ## [0.26.0] - 2022-01-24
 
 This release adds a number of improvements, most notably:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,11 +14,11 @@ The main breaking change is fairly small: [#804](https://github.com/paritytech/s
 
 - Add find last function ([#821](https://github.com/paritytech/subxt/pull/821))
 - Doc: first item is current version comment ([#817](https://github.com/paritytech/subxt/pull/817))
-- Remove sleeps from release checklist; no longer needed ([#803](https://github.com/paritytech/subxt/pull/803))
 
 ## Changed
 
 - Remove unneeded Config bounds and BlockNumber associated type ([#804](https://github.com/paritytech/subxt/pull/804))
+- Remove sleeps from release checklist; no longer needed ([#803](https://github.com/paritytech/subxt/pull/803))
 
 ## [0.26.0] - 2022-01-24
 

--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "subxt-cli"
-version = "0.26.0"
+version = "0.27.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
 publish = true
@@ -17,9 +17,9 @@ path = "src/main.rs"
 
 [dependencies]
 # perform subxt codegen
-subxt-codegen = { version = "0.26.0", path = "../codegen" }
+subxt-codegen = { version = "0.27.0", path = "../codegen" }
 # perform node compatibility
-subxt-metadata = { version = "0.26.0", path = "../metadata" }
+subxt-metadata = { version = "0.27.0", path = "../metadata" }
 # parse command line args
 clap = { version = "4.1.4", features = ["derive", "cargo"] }
 # colourful error reports

--- a/codegen/Cargo.toml
+++ b/codegen/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "subxt-codegen"
-version = "0.26.0"
+version = "0.27.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
 publish = true
@@ -21,7 +21,7 @@ proc-macro-error = "1.0.4"
 quote = "1.0.8"
 syn = "1.0.58"
 scale-info = "2.0.0"
-subxt-metadata = { version = "0.26.0", path = "../metadata" }
+subxt-metadata = { version = "0.27.0", path = "../metadata" }
 jsonrpsee = { version = "0.16.0", features = ["async-client", "client-ws-transport", "http-client"] }
 hex = "0.4.3"
 tokio = { version = "1.25", features = ["macros", "rt-multi-thread"] }

--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "subxt-examples"
-version = "0.26.0"
+version = "0.27.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
 publish = false

--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -14,10 +14,10 @@ description = "Subxt example usage"
 [dev-dependencies]
 subxt = { path = "../subxt" }
 tokio = { version = "1.25", features = ["rt-multi-thread", "macros", "time"] }
-sp-keyring = "12.0.0"
+sp-keyring = "18.0.0"
 futures = "0.3.26"
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive", "full", "bit-vec"] }
 hex = "0.4.3"
 tracing-subscriber = "0.3.11"
-sp-core = { version = "11.0.0", default-features = false }
-sp-runtime = { version = "12.0.0" }
+sp-core = { version = "16.0.0", default-features = false }
+sp-runtime = "18.0.0"

--- a/macro/Cargo.toml
+++ b/macro/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "subxt-macro"
-version = "0.26.0"
+version = "0.27.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
 publish = true
@@ -20,4 +20,4 @@ darling = "0.14.3"
 proc-macro-error = "1.0.4"
 syn = "1.0.58"
 
-subxt-codegen = { path = "../codegen", version = "0.26.0" }
+subxt-codegen = { path = "../codegen", version = "0.27.0" }

--- a/metadata/Cargo.toml
+++ b/metadata/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "subxt-metadata"
-version = "0.26.0"
+version = "0.27.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
 publish = true

--- a/subxt/Cargo.toml
+++ b/subxt/Cargo.toml
@@ -63,8 +63,8 @@ base58 = { version = "0.2.0" }
 blake2 = { version = "0.10.4", default-features = false }
 
 # These are only included is "substrate-compat" is enabled.
-sp-core = { version = "11.0.0", default-features = false, optional = true }
-sp-runtime = { version = "12.0.0", optional = true }
+sp-core = { version = "16.0.0", default-features = false, optional = true }
+sp-runtime = { version = "18.0.0", optional = true }
 
 [target.wasm32-unknown-unknown.dependencies]
 getrandom = { version = "0.2", features = ["js"] }
@@ -74,7 +74,7 @@ bitvec = "1"
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive", "full", "bit-vec"] }
 scale-info = { version = "2.0.0", features = ["bit-vec"] }
 tokio = { version = "1.25", features = ["macros", "time", "rt-multi-thread"] }
-sp-core = { version = "11.0.0", default-features = false }
-sp-runtime = { version = "12.0.0" }
-sp-keyring = "12.0.0"
-sp-version = "10.0.0"
+sp-core = { version = "16.0.0", default-features = false }
+sp-runtime = "18.0.0"
+sp-keyring = "18.0.0"
+sp-version = "16.0.0"

--- a/subxt/Cargo.toml
+++ b/subxt/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "subxt"
-version = "0.26.0"
+version = "0.27.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
 publish = true
@@ -50,8 +50,8 @@ parking_lot = "0.12.0"
 frame-metadata = "15.0.0"
 derivative = "2.2.0"
 
-subxt-macro = { version = "0.26.0", path = "../macro" }
-subxt-metadata = { version = "0.26.0", path = "../metadata" }
+subxt-macro = { version = "0.27.0", path = "../macro" }
+subxt-metadata = { version = "0.27.0", path = "../metadata" }
 
 # Provides some deserialization, types like U256/H256 and hashing impls like twox/blake256:
 impl-serde = { version = "0.4.0" }

--- a/subxt/src/tx/tx_payload.rs
+++ b/subxt/src/tx/tx_payload.rs
@@ -58,6 +58,7 @@ pub struct ValidationDetails<'a> {
 }
 
 /// This represents a statically generated transaction payload.
+#[derive(Clone, Debug)]
 pub struct StaticTxPayload<CallData> {
     pallet_name: &'static str,
     call_name: &'static str,
@@ -123,6 +124,7 @@ impl<CallData: Encode> TxPayload for StaticTxPayload<CallData> {
 }
 
 /// This represents a dynamically generated transaction payload.
+#[derive(Clone, Debug)]
 pub struct DynamicTxPayload<'a> {
     pallet_name: Cow<'a, str>,
     call_name: Cow<'a, str>,

--- a/testing/integration-tests/Cargo.toml
+++ b/testing/integration-tests/Cargo.toml
@@ -23,9 +23,9 @@ futures = "0.3.26"
 hex = "0.4.3"
 regex = "1.5.0"
 scale-info = { version = "2.0.0", features = ["bit-vec"] }
-sp-core = { version = "11.0.0", default-features = false }
-sp-keyring = "12.0.0"
-sp-runtime = "12.0.0"
+sp-core = { version = "16.0.0", default-features = false }
+sp-keyring = "18.0.0"
+sp-runtime = "18.0.0"
 syn = "1.0.0"
 subxt = { version = "0.26.0", path = "../../subxt" }
 subxt-codegen = { version = "0.26.0", path = "../../codegen" }

--- a/testing/integration-tests/Cargo.toml
+++ b/testing/integration-tests/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "integration-tests"
-version = "0.26.0"
+version = "0.27.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
 publish = false
@@ -27,8 +27,8 @@ sp-core = { version = "16.0.0", default-features = false }
 sp-keyring = "18.0.0"
 sp-runtime = "18.0.0"
 syn = "1.0.0"
-subxt = { version = "0.26.0", path = "../../subxt" }
-subxt-codegen = { version = "0.26.0", path = "../../codegen" }
+subxt = { version = "0.27.0", path = "../../subxt" }
+subxt-codegen = { version = "0.27.0", path = "../../codegen" }
 test-runtime = { path = "../test-runtime" }
 tokio = { version = "1.25", features = ["macros", "time"] }
 tracing = "0.1.34"

--- a/testing/test-runtime/Cargo.toml
+++ b/testing/test-runtime/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "test-runtime"
-version = "0.26.0"
+version = "0.27.0"
 edition = "2021"
 publish = false
 

--- a/testing/test-runtime/Cargo.toml
+++ b/testing/test-runtime/Cargo.toml
@@ -6,7 +6,7 @@ publish = false
 
 [dependencies]
 subxt = { path = "../../subxt" }
-sp-runtime = "12.0.0"
+sp-runtime = "18.0.0"
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive", "full", "bit-vec"] }
 
 [build-dependencies]

--- a/testing/ui-tests/Cargo.toml
+++ b/testing/ui-tests/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "ui-tests"
-version = "0.26.0"
+version = "0.27.0"
 edition = "2021"
 publish = false
 

